### ISSUE TITLE
etcdserver/api/v3rpc/watch.go panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by sonyafenge** [futurewei-cloud/arktos#737]
**What happened**:
started 5K runs using community ETCD3.4.4 with workload-controller disalbed and also PR: https://github.com/futurewei-cloud/arktos/pull/709 included, got the panic below:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8a59ae]

goroutine 28159312 [running]:
go.etcd.io/bbolt.(*Tx).Bucket(...)
        /home/ubuntu/go/pkg/mod/go.etcd.io/bbolt@v1.3.3/tx.go:101
go.etcd.io/etcd/mvcc/backend.(*concurrentReadTx).UnsafeRange(0xc01ff8ffb0, 0x1a4a06a, 0x3, 0x3, 0xc1393d6640, 0x11, 0x12, 0x0, 0x0, 0x0, ...)
        /home/ubuntu/go/src/etcd/mvcc/backend/read_tx.go:195 +0x67e
go.etcd.io/etcd/mvcc.(*storeTxnRead).rangeKeys(0xc00b6b0000, 0xc09a84da70, 0x23, 0x30, 0x0, 0x0, 0x0, 0x12de4f2, 0x0, 0x12dd142, ...)
        /home/ubuntu/go/src/etcd/mvcc/kvstore_txn.go:147 +0x29e
go.etcd.io/etcd/mvcc.(*storeTxnRead).Range(0xc00b6b0000, 0xc09a84da70, 0x23, 0x30, 0x0, 0x0, 0x0, 0x0, 0x12dd142, 0x8a4400, ...)
        /home/ubuntu/go/src/etcd/mvcc/kvstore_txn.go:51 +0xb0
go.etcd.io/etcd/mvcc.(*metricsTxnWrite).Range(0xc00b6b0030, 0xc09a84da70, 0x23, 0x30, 0x0, 0x0, 0x0, 0x0, 0x12dd142, 0xfdf600, ...)
        /home/ubuntu/go/src/etcd/mvcc/metrics_txn.go:37 +0xaf
go.etcd.io/etcd/mvcc.(*readView).Range(0xc0002dc020, 0xc09a84da70, 0x23, 0x30, 0x0, 0x0, 0x0, 0x0, 0x12dd142, 0x2000, ...)
        /home/ubuntu/go/src/etcd/mvcc/kv_view.go:39 +0x125
go.etcd.io/etcd/etcdserver/api/v3rpc.(*serverWatchStream).sendLoop(0xc0b8766300)
        /home/ubuntu/go/src/etcd/etcdserver/api/v3rpc/watch.go:377 +0x1da
go.etcd.io/etcd/etcdserver/api/v3rpc.(*watchServer).Watch.func1(0xc0b8766300)
        /home/ubuntu/go/src/etcd/etcdserver/api/v3rpc/watch.go:163 +0x2b
created by go.etcd.io/etcd/etcdserver/api/v3rpc.(*watchServer).Watch
        /home/ubuntu/go/src/etcd/etcdserver/api/v3rpc/watch.go:162 +0x275
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
![image](https://user-images.githubusercontent.com/45471193/93382559-1f393300-f817-11ea-893c-022953e2b4e1.png)

**Environment**:
- Arktos version (use `kubectl version`):
```
commit f1db91c31e4f650e2a00dfc753d92fbc65212ab4
Author: sonyafenge <sonyafenge@gmail.com>
Date:   Fri Aug 28 21:32:26 2020 +0000

    disable workload-controller-manager

commit 4a7fb24b2df0d7908917bc5e309272cbddb5f525
Author: sonyafenge <sonyafenge@gmail.com>
Date:   Wed Sep 9 02:51:40 2020 +0000

    ETCD3.4.4

commit 0ab02fe96d178c69c0d23954a2f8de81ab9b6538
Author: Ying Huang <sindica2000@yahoo.com>
Date:   Mon Sep 14 15:58:05 2020 +0000

    UT for aggregated watch

commit 553aaaa028ee8353130fe2bb6eec04c47e94b004
Author: Ying Huang <sindica2000@yahoo.com>
Date:   Sat Sep 12 13:38:27 2020 +0000

    Close aggregated watcher upon connection refused error.

commit 447568205ede15153ce7287cf4f553ba3605908f
Author: Ying Huang <sindica2000@yahoo.com>
Date:   Sat Sep 12 13:13:20 2020 +0000

    Catch and log/recover from aggregated watch

commit 7f59296c6dac4402d43fbb39584937cff1bf4e83
Author: Vinay Kulkarni <vinaykul@users.noreply.github.com>
Date:   Tue Sep 15 16:13:50 2020 -0700

    Initialize WorkloadInfo during defaulting (#706)

    * Initialize WorkloadInfo during defaulting

    * Enable WorkloadInfo initialization via feature-gate
```
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by Sindica** [comment]
I would suggest using ETCD version used in community run:
etcd Version: 3.4.3
Git SHA: 3cf2f69b5
Go Version: go1.12.12

**Comment by Sindica** [comment]
I found the panic in sonya-uswest2:/home/sonyali/logs/perf-test/gce-5000/arktos/0915-issue645-3a0w1e but not in sonya-useast1:/home/sonyali/logs/perf-test/gce-5000/arktos/0915-baseline-1a0w1e

**Comment by Sindica** [comment]
From etcd issue https://github.com/etcd-io/etcd/pull/11437
It is seems to be a problem when shutting down gRPC server
The issue was detected in etcd unit test and fixed in this commit: https://github.com/etcd-io/etcd/commit/d70251835d460587b8fe377188132560bc205fc3

Even though this commit was merged on Dec. 11th, 2019, it is only in master. Not merged into release-3.4 branch.

**Comment by Sindica** [comment]
Closing as we cannot do anything with it now. It should happen only when shutting down gRPC server, which could means ETCD is not healthy.
